lib/core/elastic_connector_actions.rb

```diff
@@ -117,9 +118,31 @@ def client
         @client ||= Utility::EsClient.new
       end
 
+      # should only be used in CLI
+      def ensure_content_index_exists(index_name, use_icu_locale = false, language_code = nil)
+        settings = Utility::Elasticsearch::Index::TextAnalysisSettings.new(:language_code => language_code, :analysis_icu => use_icu_locale).to_h
+        mappings = Utility::Elasticsearch::Index::Mappings.default_text_fields_mappings(:connectors_index => true)
+
+        body_payload = { settings: settings, mappings: mappings }
+        ensure_index_exists(index_name, body_payload)
+      end
+
       # should only be used in CLI
       def ensure_index_exists(index_name, body = {})
-        client.indices.create(:index => index_name, :body => body) unless client.indices.exists?(:index => index_name)
+        if client.indices.exists?(:index => index_name)
```

Comment:
I think we should move this to a separate method `ensure_content_index_exists` because this logic only applies to the content index.

`ensure_index_exists` should serve a general purpose.